vendor/zkllvm-metacraft-circuits/src/circuit_utils/circuit_byte_utils.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "base_types.h"
+
+using namespace nil::crypto3;
+
+namespace circuit_byte_utils {
+
+    using sha256_t = typename hashes::sha2<256>::block_type;
+
+    bool is_same(sha256_t block0, sha256_t block1) {
+
+        bool result = true;
+        for (auto i = 0; i < sizeof(block0) / sizeof(block0[0]) && result; i++) {
```

Comment:
implement a sizeof function that accepts an std::array and deduces its type and size and returns back the size